staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -133,8 +134,12 @@ func (w *watcher) Watch(ctx context.Context, key string, rev int64, recursive bo
 	if err != nil {
 		return nil, err
 	}
+	shard, err := genericapirequest.ValidShardFrom(ctx)
```

Comment:
what does this return if there is no shard in the context?

Reply:
```
// ValidShardFrom returns the value of the shard key from the context.
// An empty value is returned if there's no shard key,
// or if the shard name is empty, and it wasn't a wildcard request.
```